modules/exploits/linux/http/primefaces_weak_encryption_rce.rb

```diff
@@ -0,0 +1,141 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Primefaces Remote Code Execution Exploit',
+        'Description' => %q{
+          This module exploits an expression language remote code execution flaw in the Primefaces JSF framework.
+          Primefaces versions prior to 5.2.21, 5.3.8 or 6.0 are vulnerable to a padding oracle attack,
+          due to the use of weak crypto and default encryption password and salt.
+
+          Tested against Docker image with Tomcat 7.0 with the Primefaces 5.2 showcase application. See
+          documentation for working payloads.
+        },
+        'Author' => [
+          'Bjoern Schuette', # EDB
+          'h00die' # lots of fixes, documentation, standardization
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2017-1000486'],
+          ['URL', 'https://blog.mindedsecurity.com/2016/02/rce-in-oracle-netbeans-opensource.html'],
+          ['URL', 'https://web.archive.org/web/20180515174733/https://cryptosense.com/blog/weak-encryption-flaw-in-primefaces'],
+          ['URL', 'http://schuette.se/2018/01/16/in-your-primeface/'],
+          ['URL', 'https://github.com/primefaces/primefaces/issues/1152'],
+          ['URL', 'https://github.com/pimps/CVE-2017-1000486/tree/master'],
+          ['EDB', '43733']
+        ],
+        'Payload' => {
+          'BadChars' => '"\'\\' # all threw errors
+        },
+        'Privileged' => true,
+        'DisclosureDate' => '2016-02-15',
+        'Platform' => ['unix', 'bsd', 'linux', 'osx', 'win'],
```

Comment:
Since this includes every platform, can we move the exploit from `exploits/linux` to `exploits/multi`?